hooks/field/useSelectField.ts

```diff
@@ -286,6 +277,17 @@ export function useSelectField<T> (
         ]
     );
 
+    const updateSearchFieldCallback = useCallback(
+        () => {
+            if( searchField ) setSearchField('')
+        },
+        [
+            updateCurrentItemIndexCallback,
```

Comment:
Why do you set these as the dependency when they aren't used inside the callback?

Also why is there no dependency for `searchField` and `setSearchField`...

Also probably better would be to use `setSearchField((value) => value ? value : '')` -- that way you don't need the if clause or dependency for the current value, and it works asynchronously correctly (maybe).

Reply:
This callback is only used to nullify the SearchField after user selects an item, so that full list of items can be seen afterwards.

Currently this is the only place where SearchField is emptied, if I set SearchField as dependency then the field nullifies itself everytime when someone starts typing, thus filtering doesn't work.

### Ofcourse the Callback function itself is quite pointless in this case, the setSearchField could be called from functions that I use as dependency, but in that case the logic is spread around instead of getting called from one specific place.